### DISCUSSION TITLE
Mutter vs Xfwm4

### DISCUSSION THREAD

**Discussion by edwardp3** [lxqt/lxqt#2410] [General]
Wanted to try another window compositor (for transparent menus) and installed Mutter (on Debian Sid). It works with LXQt 1.2.0, however, most of the LXQt themes installed (from both the official packages and also User Themes downloaded from this site and Pling), did not display transparent menus. Out of the LXQt official themes, only Dark, Debian, Kvantum and Valendas, displayed transparent menus with Mutter as the Window Manager. None of the User Themes displayed it.

Xfwm4 displayed transparent menus for all of the LXQt official themes, as well as all the User Themes.

I also wanted to check the Desktop Switcher, as there appears to be a bug when Xfwm4 is used. It will display more than 2, if the setting is changed during the current session, but on a new login, it only displays 2.

_As previously mentioned here, kwin is too slow for my desktops._

**Comment by tsujan** [comment]
> As previously mentioned here, kwin is too slow for my desktops.

Before you use KWin, make sure that no compositor (picom, compton,...) is auto-started or running. I've encountered a few reports about "slow KWin" whose cause was only that users had another compositor running alongside KWin.

Apart from that, NVIDIA+KWin may have problems.

**Reply by edwardp3** [reply]
When I tried Kwin, the only other compositors/WM's installed were Openbox and Xfwm4. I switched it to Kwin in Session Settings, then logged out/login. The desktop I tried this on is now 12 years old, but it has integrated AMD graphics (Radeon RS780L).

Plus, Synaptic also wanted to install at least 50 other packages, in addition to Kwin. Also had some issues with Picom.

**Reply by tsujan** [reply]
> Plus, Synaptic also wanted to install at least 50 other packages, in addition to Kwin

That's determined by which packages you've already installed, how your distro splits packages (or doesn't), and how it manages dependencies. On Arch, `mutter` depends on 17 packages, `kwin` depends on 9, `xfwm4` on 6, and `openbox` on 9.

**Reply by edwardp3** [reply]
Debian apparently installed most of what was otherwise needed. Installing mutter, also installed `gnome-settings-daemon-common`, `libmutter-11-0` and `mutter-common`. Same with picom, in addition, `libconfig9` and `libev4`. With picom, the same four themes showing transparent menus with mutter, displayed them with picom. So from what I can see, there's not much difference between the two of them. Plus with picom, the LXQt desktop appearing after a login, is delayed for several seconds vs mutter and xfwm4. xfwm4 is the best out of the three, IMO.
